pkg/controller/replicaset/replica_set.go

```diff
@@ -802,10 +804,28 @@ func getPodsToDelete(filteredPods, relatedPods []*v1.Pod, diff int) []*v1.Pod {
 	if diff < len(filteredPods) {
 		podsWithRanks := getPodsRankedByRelatedPodsOnSameNode(filteredPods, relatedPods)
 		sort.Sort(podsWithRanks)
+		filteredPods = podsWithRanks.Pods
```

Comment:
it doesn't seem like this line is necessary.

Reply:
I actually had a question about this line. Looking at the whole function, it seems like nothing is ever actually done with the sorted list of podRanks: https://github.com/kubernetes/kubernetes/blob/master/pkg/controller/replicaset/replica_set.go#L799-L807

This is after [this change](https://github.com/kubernetes/kubernetes/commit/980b6406b24b41bc0ef061624385f1f420708819#diff-f658ce5cda23ed9920c4ee4eaaee968fb5a4dbd4a0ccc7f478db399f2b182380R762-R763), which added the call to `getPodsRankedByRelatedPodsOnSameNode`.

Prior to that, `sort()` was called directly on `filteredPods`: https://github.com/kubernetes/kubernetes/blame/865c3c56706d131c3164d16812fec58454a77020/pkg/controller/replicaset/replica_set.go#L684-L694

I asked @soltysh about this, and he wasn't sure either (though it seems odd that this sort of bug would go so long without being noticed).

Do you have any idea how the new `podsWithRanks` variable declared on L805 gets passed to the return statement at the bottom of the function?

Reply:
Ah maybe it is because `filteredPods` is the underlying slice in the returned `ActivePodsToRank` struct: https://github.com/kubernetes/kubernetes/blob/d8aad7944cc71314913fc0b43d6731e2c36f4928/pkg/controller/replicaset/replica_set.go#L813-L825

So calling `sort()` on the returned struct will modify `filteredPods` too?

Reply:
correct :)